cmd/snapd-env-generator/main.c

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include<stdlib.h>
+#include<string.h>
+#include<stdio.h>
+
+#include "config.h"
+
+// Systemd environment generators work since version 233 which ships
+// in Ubuntu 17.10+
+int main(int argc, char **argv)
+{
+        const char *snap_bin_dir = SNAP_MOUNT_DIR"/bin";
```

Comment:
Didn't notice it before and indent will probably fix it, but it should be `SNAP_MOUNT_DIR "/bin"` (gcc used to yell about that).